ncm-spma/src/main/pan/components/spma/functions.pan

```diff
@@ -101,8 +101,10 @@ function resolve_pkg_rep = {
     arg = repository list
 }
 function purge_rep_list = {
-    foreach (i;rep;SELF) {
-        rep['contents'] = null;
+    if (is_list(SELF)) {
```

Comment:
The primary trigger for this is that the repository list is undefined when running the unit-tests.